src/lib/y2storage/filesystems/btrfs.rb

```diff
@@ -0,0 +1,131 @@
+      #
+      #   @note When a new Btrfs filesystem is created, a top level subvolume is added and
+      #   initialized as default subvolume.
+      #
+      #   @see #default_btrfs_subvolume
+      #
+      #   @return [BtrfsSubvolume]
+      storage_forward :default_btrfs_subvolume, as: "BtrfsSubvolume"
+
+      # @!method default_btrfs_subvolume=(subvolume)
+      #   Set the default Btrfs subvolume
+      #
+      #   @param subvolume [BtrfsSubvolume]
+      storage_forward :default_btrfs_subvolume=
+
+      # @!method btrfs_subvolumes
+      #   Collection of Btrfs subvolumes of the filesystem
+      #   @return [Array<BtrfsSubvolumes>]
+      storage_forward :btrfs_subvolumes, as: "BtrfsSubvolume"
+
+      # @!method find_btrfs_subvolume_by_path(path)
+      #   Finds a subvolume by its path
+      #
+      #   @param path [String] subvolume path
+      #   @return [BtrfsSubvolume, nil] nil if it does not find a subvolume with this path
+      storage_forward :find_btrfs_subvolume_by_path, as: "BtrfsSubvolume"
+
+      # Return the default subvolume, creating it when necessary
+      #
+      # If a specific default subvolume path is requested, create it; if not,
+      # use the toplevel subvolume that is implicitly created by mkfs.btrfs.
+      #
+      # This default subvolume is the parent for all others on Btrfs 'filesystem'.
+      #
+      # @param path [String, nil] path for the default subvolume
+      #
+      # @return [BtrfsSubvolume]
+      def get_or_create_default_btrfs_subvolume(path: nil)
+        # The toplevel subvolume is implicitly created by mkfs.btrfs.
+        # It does not have a name, and its subvolume ID is always 5.
+        top_level = top_level_btrfs_subvolume
+        if path && !path.empty?
+          # If the "@" subvolume is specified in control.xml, this must be
+          # created first, and it will be the parent of all the other
+          # subvolumes. Otherwise, the toplevel subvolume is their direct parent.
+          # Notice that this "@" subvolume does not show up in "btrfs subvolume
+          # list".
+          default_subvolume = top_level.create_btrfs_subvolume(path)
+          default_subvolume.set_default_btrfs_subvolume
+        else
+          # Top level is set as default
+          default_subvolume = top_level
+        end
+
+        default_subvolume
+      end
+
+      # Deletes a btrfs subvolume that belongs to the filesystem
+      #
+      # @param devicegraph [Devicegraph]
+      # @param path [String] path of subvolume to delete
+      def delete_btrfs_subvolume(devicegraph, path)
+        subvolume = find_btrfs_subvolume_by_path(path)
+        return if subvolume.nil?
+
+        subvolume.remove_descendants
+        devicegraph.to_storage_value.remove_device(subvolume.to_storage_value)
+      end
+
+      # The path that a new default btrfs subvolume should have
+      #
+      # @return [String]
+      def self.default_btrfs_subvolume_path
```

Comment:
This returns "" if the value is indeed "" (meaning the top level subvolume) and if the value is not there. Just asking - wouldn't it be better to return nil if the value is not specified?

I don't think is relevant in general, but it would make more sense when compared to the `#ensure_default_btrfs_subvolume` (which really makes a difference between nil and "").

As said, not a big deal in reality. Check `#ProposalSettings#product_feature` to see how to differentiate nil and "".